cns/restserver/api.go

```diff
@@ -1522,3 +1523,50 @@ func (service *HTTPRestService) nmAgentSupportedApisHandler(w http.ResponseWrite
 
 	logger.Response(service.Name, nmAgentSupportedApisResponse, resp.ReturnCode, serviceErr)
 }
+
+// getVMUniqueID retrieves VMUniqueID from the IMDS
+func (service *HTTPRestService) getVMUniqueID(w http.ResponseWriter, r *http.Request) {
+	logger.Printf("[Azure CNS] getVMUniqueID")
+	logger.Request(service.Name, "getVMUniqueID", nil)
+	ctx := r.Context()
+
+	switch r.Method {
+	case http.MethodGet:
+		imdsClient := getIMDSClient(r)
```

Comment:
I think service should have a member `IMDSClient`, which is an interface defined as 

```golang
type IMDSClient interface {
    func GetVMUniqueID(ctx context.Context) (string, error)
}
````
Then, you would simply mock the IMDSClient in your unit tests by implementing a mock of your IMDSClient interface in your UT (say, `mockIMDSCli`) and setting `service.IMDSClient = &mockIMDSCli{}`. Then you don't need to pass along a port as an http header, and you treat the existing IMDSClient package like an external, third-party library (which is good for separation of concerns)

Reply:
Thanks. I added up introducing mock IMDS client.